drivers/ethernet/eth_xmc4xxx.c

```diff
@@ -792,6 +793,30 @@ static inline int eth_xmc4xxx_init_timestamp_control_reg(ETH_GLOBAL_TypeDef *reg
 	return 0;
 }
 
+static int eth_xmc4xxx_get_mac_address_eeprom(const struct device *dev)
+{
+#if DT_INST_NODE_HAS_PROP(0, mac_eeprom)
+	struct eth_xmc4xxx_data *dev_data = dev->data;
+	const struct device *eeprom_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, mac_eeprom));
+	int ret;
+
+	if (!device_is_ready(eeprom_dev)) {
+		LOG_ERR("EEPROM device not ready.");
+		return -ENODEV;
+	}
+
+	ret = eeprom_read(eeprom_dev, CONFIG_ETH_XMC4XXX_MAC_EEPROM_ADDRESS, dev_data->mac_addr,
+			  sizeof(dev_data->mac_addr));
+	if (ret < 0) {
+		LOG_ERR("Error reading MAC address from EEPROM [%d]", ret);
+	}
+
+	return ret;
+#else
+	return -ENODEV;
+#endif
+}
```

Comment:
Put the compile check around the function.

```c
#if DT_INST_NODE_HAS_PROP(0, mac_eeprom)
static int eth_xmc4xxx_get_mac_address_eeprom(const struct device *dev)
{
	struct eth_xmc4xxx_data *dev_data = dev->data;
	const struct device *eeprom_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, mac_eeprom));
	int ret;

	if (!device_is_ready(eeprom_dev)) {
		LOG_ERR("EEPROM device not ready.");
		return -ENODEV;
	}

	ret = eeprom_read(eeprom_dev, CONFIG_ETH_XMC4XXX_MAC_EEPROM_ADDRESS, dev_data->mac_addr,
			  sizeof(dev_data->mac_addr));
	if (ret < 0) {
		LOG_ERR("Error reading MAC address from EEPROM [%d]", ret);
	}

	return ret;
}
#endif
```

Reply:
In my patch returning -ENODEV makes using the function cleaner and the compiler optimizes out the function when the node is not in the devicetree.